src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -305,11 +305,11 @@ static ImmutableArray<FunctionPointerParameterSymbol> makeParametersFromMetadata
 
             static RefKind getRefKind(ParamInfo<TypeSymbol> param, ImmutableArray<CustomModifier> paramRefCustomMods, RefKind hasInRefKind)
             {
-                // PROTOTYPE(func-ptr): Need to encode out params as a custom modifier of some kind
                 return param.IsByRef switch
                 {
                     false => RefKind.None,
                     true when CustomModifierUtils.HasInAttributeModifier(paramRefCustomMods) => hasInRefKind,
+                    true when CustomModifierUtils.HasOutAttributeModifier(paramRefCustomMods) => RefKind.Out,
```

Comment:
> true when CustomModifierUtils.HasOutAttributeModifier(paramRefCustomMods) => RefKind.Out, [](start = 19, length = 90)

This doesn't look like the right thing to do for a return RefKind. Please add a test

Reply:
I'll add a test, but I'll also add an assert. The return parameter is decoded only allowing `InAttribute`, so if it encountered an out during decode we'd throw and never construct a function pointer type symbol at all.

---
In reply to: [409789040](https://github.com/dotnet/roslyn/pull/43355#discussion_r409789040) [](ancestors = 409789040)